pymatgen/io/vasp/outputs.py

```diff
@@ -1957,8 +1957,8 @@ def __init__(self, filename):
         else:
             mag = mag_x
 
-        # data from beginning of OUTCAR
-        run_stats["cores"] = None
+        # Data from beginning of OUTCAR
+        run_stats["cores"]  = -1
```

Comment:
I changed the start value for number of cores from `None` to `-1` for type compatibility, let me know if this could potentially break anything, or there is any better way.

Similarly from memory usage:
```python
                if time_patt.search(line):
                    tok = line.strip().split(":")
                    try:
                        # try-catch because VASP 6.2.0 may print
                        # Average memory used (kb):          N/A
                        # which cannot be parsed as float
                        run_stats[tok[0].strip()] = float(tok[1].strip())
                    except ValueError:
                        run_stats[tok[0].strip()] = -1
                    continue
```

Reply:
i think it's safer to stick with `run_stats["cores"] = None`. can we not update the type to `int | None` and explicitly guard against `None` where necessary?

Reply:
Yes! Looks we could allow `None` by explicitly set type as `run_stats: dict[str, float | None] = {}`

Changed in 16d26857a6f7f9a433001f9048bd692e932e88a4